components/buildList/js/buildList.controller.js

```diff
@@ -6,6 +6,464 @@ buildApp.controller('buildListController', ['$scope', '$q', '$interval', 'buildL
+	 * @param  {Object} firewallDefered
+	 */
+	function updateBuild(build, buildDefered) {
+
+		if (build.status === 'complete' || build.status === 'rejected') {
+			buildDefered.resolve();
+			return;
+		}
+
+		var firewallsPromises = [];
+
+		angular.forEach(build.firewalls, function(firewall) {
+			var firewallDefered = $q.defer();
+			firewallsPromises.push(firewallDefered.promise);
+			updateFirewall(firewall, firewallDefered); 
+		});
+
+		//update build information after all its firewalls have been updated
+		$q.all(firewallsPromises).then(function() {
+			var fStates = [];
+
+			var buildJobs = {
+				metrics: {
+					percentage: 0,
+					status: []
+				},
+				build: {
+					percentage: 0,
+					status: []
+				},
+				unitTest: {
+					percentage: 0,
+					status: []
+				},
+				functionalTest: {
+					percentage: 0,
+					status: []
+				}
+			};
+
+			angular.forEach(build.firewalls, function(firewall) {
+				fStates.push (firewall.status);
+				angular.forEach(firewall.jobs, function(job, jobName) {
+					buildJobs[jobName].percentage += job.percentage;
+					buildJobs[jobName].status.push(job.status);
+				});
+			});
+
+			angular.forEach(buildJobs, function(value) {
+				value.percentage = parseInt(value.percentage / build.firewalls.length, 10);
+				value.status = setStatusOfProcess(value.status);
+			});
+
+			build.jobs = buildJobs;
+			build.status = setStatusOfProcess(fStates);
+			buildDefered.resolve();
+		});
+	}
+
+
+	/**
+	 * Starts to calculate the states of the builds.
+	 * @method updateBuildsList
+	 */
+	function updateBuildsList() {
+
+		timeElapsed += intervalTime;
+		var overallStatus = 'running';
+		var buildPromises = [];
+
+		angular.forEach($scope.buildList, function(build) {
+			var buildDefered = $q.defer();
+			buildPromises.push(buildDefered.promise);
+			updateBuild(build, buildDefered);
+		});
+
+		$q.all(buildPromises).then(function() {
+			var buildStates = [];
+
+			angular.forEach($scope.buildList, function(build) {
+				buildStates.push (build.status);
+			});
+
+			overallStatus = setStatusOfProcess(buildStates);
+			if (overallStatus === 'complete' || overallStatus === 'rejected') {
+				$interval.cancel(updateBuildsInterval);
+			}
+		});
+	}
+
+	init();
+}]).controller('buildListController', ['$scope', '$q', '$interval', 'buildListService', function ($scope, $q, $interval, buildListService) {
+	
+	var deferred = $q.defer(),
+		updateBuildsInterval,
+		intervalTime = 5000,
+		timeElapsed = 0;
+
+	/**
+	 * Initializes the functionality of this controller
+	 *  - requests the data
+	 *  - starts to calculate all the processes
+	 * @method init
+	 */
+	function init() {
+		getData();
+		$q.when(deferred.promise, function() {
+			updateBuildsInterval = $interval(updateBuildsList, intervalTime);
+		});
+	}
+
+	/**
+	 * Requests for the build list data
+	 * @method getData
+	 */
+	function getData() {
+		return buildListService.getData().then(function(response) { 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3ATrailingWhitespace)